oauth2client/pkce.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2016 Google Inc. All rights reserved.
```

Comment:
> Ultimately, I think it's kind of meaningless because people can still import it even if it's named with an underscore.

Regardless, this is an accepted Python practice and this library follows that practice. Things that we do not intended for users to use should be private and noted with an `_`.

> I don't think it's necessary to scare users away from some possibly-useful utility functions.

The functions in question are useless outside of the context of using with with OAuth2WebServerFlow. Possibly useful isn't justification enough for public API surface. In open-source, no is temporary and yes is forever.

> If you don't want to commit to maintaining those functions, don't document them?

I'd prefer we both documented them and followed the practice of marking them private with an underscore.

> If everything internal to the operation of this library is supposed to be 'private'

This library has lots of warts and I'm sure you've noticed we're putting a lot of effort into fixing them. To help that spirit, we encourage new contributions to do as we say, not as we've done.